stream/distributedlog/core/src/main/java/org/apache/distributedlog/EnvelopedEntryWriter.java

```diff
@@ -165,7 +165,7 @@ public synchronized ByteBuf getBuffer() throws InvalidEnvelopedEntryException, I
         if (null == finalizedBuffer) {
             finalizedBuffer = finalizeBuffer();
         }
-        return finalizedBuffer.slice();
+        return finalizedBuffer.retainedSlice();
```

Comment:
This is changing the net counts for this buffer. I don't really know much on this code, but would this still be correct when `envelopeBeforeTransmit` is false?

Reply:
@merlimat I am not familiar with dlog's code. 
This mimicked what the tests expected/extra retain from addAsync so I figured it cannot be worse that it was initially.

I hope @sijie can help with the question on expectations from envelopeBeforeTransmit